### PR TITLE
Show contacts on raise repair form

### PR DESCRIPTION
**Pull request by NeilvB**
### Description of change

Add resident contact information to the base of the raise repairs form.

Redact sensitive information for responses from the Node API which are not for agents as they have no need to see this.

 ### Before Merge

 - [x] Double-check with backend if JSON structure they return is correct - should contacts be a nested object?

### Story Link

https://trello.com/c/ckaBKNzB/345-as-an-rcc-agent-i-can-view-the-contact-name-and-number-assigned-to-a-property-when-raising-a-repair-so-i-can-check-details-if-ne

### Screenshots

These are from the base of the raise repairs form.

image

image

### CODE REVIEW

File: src/components/Property/Contacts/ContactsNoneAvailable.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const ContactsNoneAvailable = () => {
```
Comment (benbaumann95) [inline src/components/Property/Contacts/ContactsNoneAvailable.js:1]: Wonder if we can make this component more reusable such as `WarningText` and then pass the string value as props. Seems like nothing is custom apart from the text.

The reason I raise this is that one of the stories coming up will need to use the same warning text template. But happy to address this and refactor when the time comes

File: src/components/Property/Contacts/ContactsTable.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import PropTypes from 'prop-types'
+import ContactsRow from './ContactsRow'
+
+const ContactsTable = ({ contacts }) => {
+  return (
+    <table className="govuk-table">
```
Comment (benbaumann95) [inline src/components/Property/Contacts/ContactsTable.js:6]: You may have missed it but Andrew has added some reusable <Table> <Thead> <TBody> components etc. Going forward might be worth using this (so we don't have to repeat all the generic classnames) but I know older table components don't have this updated yet so could address as one PR later

File: cypress/fixtures/properties/property.json [hunk 3]
```diff
@@ -41,5 +41,19 @@
   "tenure": {
     "typeCode": "SEC",
     "typeDescription": "Secure"
+  },
+  "contacts": {
```
Comment (benbaumann95) [inline cypress/fixtures/properties/property.json:45]: I've seen your comments to propose a change to this nesting and totally agree that this makes no sense like this

File: cypress/integration/raise-repair/form.spec.js [hunk 4]
```diff
@@ -55,6 +56,14 @@ describe('Raise repair form', () => {
       ]
     )
 
+    // Property contact information table
+    cy.get('.govuk-table__caption').contains('Contacts')
+    cy.get('.govuk-table__row .govuk-table__cell').contains('Mark Gardner')
+    cy.get('.govuk-table__row .govuk-table__cell').contains('00000111111')
+    cy.get('.govuk-table__row .govuk-table__cell').contains('00000222222')
+    cy.get('.govuk-table__row .govuk-table__cell').contains('Luam Berhane')
+    cy.get('.govuk-table__row .govuk-table__cell').contains('00000666666')
```
Comment (benbaumann95) [inline cypress/integration/raise-repair/form.spec.js:65]: Is it possible to add a data-attribute to identify the rows exactly? Maybe a bit overkill but I find it useful when doing tests on multiple records to make sure i'm targeting the correct index/row etc

**Review by benbaumann95 — APPROVED** [review]
Left some minor thoughts but not blocking merging

File: cypress/fixtures/properties/property.json (hunk 3 above)
Comment (NeilvB) [reply cypress/fixtures/properties/property.json:45]: Backend has been updated, and frontend matched up here: https://github.com/LBHackney-IT/repairs-hub-frontend/pull/89/commits/499c4f19c4bdf645eb77c6893bbee9a5e72a5bfb

File: src/components/Property/Contacts/ContactsTable.js (hunk 2 above)
Comment (NeilvB) [reply src/components/Property/Contacts/ContactsTable.js:6]: Thanks. I'm going to leave this for now and I think we can address as a sweep later.

File: cypress/integration/raise-repair/form.spec.js (hunk 4 above)
Comment (NeilvB) [reply cypress/integration/raise-repair/form.spec.js]: Thanks. I definitely agree that these should be more focused. I was interested to read about the Cypress recommendation to use data attributes just for testing.

For this case I think I'm mostly interested in the numbered row in the table and I hoped to just rely on the DOM ordering. I've chosen to go for using `cy.eq()` to scope the query to the first and then the second row. I feel a little resistant to adding data attributes to our markup if they don't support the functioning or presentation of the app, but I can also see how that would be super useful for testing too especially as we're using lots of generic classes.

I'm going to go ahead with this for now as it is an improvement hopefully, but happy and interested to talk about this further.

```
    cy.get('.govuk-table')
      .contains('Contacts')
      .parent()     // grab the parent table for scoping the following
      .within(() => {
        cy.get('tbody>tr').eq(0).contains('Mark Gardner')   // first row
        cy.get('tbody>tr').eq(0).contains('00000111111')
        cy.get('tbody>tr').eq(0).contains('00000222222')

        cy.get('tbody>tr').eq(1).contains('Luam Berhane')  // second row
        cy.get('tbody>tr').eq(1).contains('00000666666')
      })
```

File: src/components/Property/Contacts/ContactsNoneAvailable.js (hunk 1 above)
Comment (NeilvB) [reply src/components/Property/Contacts/ContactsNoneAvailable.js]: Great shout. I've extracted a `WarningText` here: https://github.com/LBHackney-IT/repairs-hub-frontend/pull/89/commits/aa696b0ae5ed1cee3dcfe9dd037f37fbdc078c81

**Pull request merged**